src/DotVVM.Framework/Runtime/Caching/IDotvvmCache.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DotVVM.Framework.Runtime.Caching
+{
+    public interface IDotvvmCache
+    {
+        T GetOrAdd<T>(object key, Func<object, DotvvmCachedItem<T>> factoryFunc);
+
+        T Get<T>(object key);
+
+        object Remove(object key);
+    }
+
+    public class DotvvmCachedItem<T>
+    {
+        public DotvvmCachedItem(T value, DotvvmCacheItemPriority priority)
+        {
+            this.Value = value;
+            Priority = priority;
+        }
+
+        public DotvvmCachedItem(T value)
```

Comment:
Why not one constructor with default parameter? Then it immediately obvious what are the default and which parameters can be used.